actions/parse-secrets-definitions/main.py

```diff
@@ -0,0 +1,57 @@
+import os
+import re
+
+import typer
+from typing_extensions import Annotated
+
+DEFAULT_DELIMITER = "!!!"
+
+# CAVEAT: will only work for one project at a time
+# might be problematic if we want to inject secrets from multiple projects
+
+# CAVEAT: this script can produce lists of secrets that are not valid, e.g. if the secret name is empty or the same secret is referenced multiple times
```

Comment:
Ideally we can check for those cases, so that parsing always works correctly

Reply:
Added a comment for the multiple projects case -> just run the action another time
Added a set for deduping secrets and a Value error for empty secrets